src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2462,11 +2469,13 @@ private void visitObjectType(
       indent();
       emitBreak();
       // Prevent accidental structural typing - emit every class with a private field.
-      if (type.isNominalConstructor()
-          && !type.isInterface()
-          // But only for non-extending classes (TypeScript doesn't like overriding private fields)
-          && getSuperType(type) == null) {
-        emit("private noStructuralTyping_: any;");
+      if (type.isNominalConstructor() && !type.isInterface()) {
+        emit("private noStructuralTyping_");
+        // TypeScript does not allow overriding private properties from superclasses, so make sure
+        // to emit a private property name that's specific to this type.
+        String suffix = type.hasDisplayName() ? escapeForJSProperty(type.getDisplayName()) : "";
```

Comment:
What does it mean if the type doesn't have a display name?  Maybe we shouldn't behave nominally in this case?

Reply:
I cannot construct a test case where we emit a Clutz declaration for a type, but the type has no display name. Any type we reach to emit, we reach through the global symbol table, and anything in that table by definition has a reachable name, which Closure uses as the display name.

So this is just defensive coding, but I think it cannot happen in practice (I know, famous last words...).